platform/cxxsupport/mstd_type_traits

```diff
@@ -1200,15 +286,6 @@ using remove_cvref_t = typename remove_cvref<T>::type;
 
 }
 
-#ifndef __CC_ARM
```

Comment:
This code should have been kept.

Reply:
ARMC6 and GCC_ARM toolchain, This is always true, so it is not required